### PR TITLE
CRD create / update at reconcile

### PR DESCRIPTION
**Pull request by JoseAlban**
Re. #71 .
This ensures that CRDs (shipcatConfig and shipcatManifest) are in the correct state when attempting a `shipcat cluster crd reconcile`.

Tested manually on a local cluster.

### CODE REVIEW

File: shipcat_definitions/src/crds.rs [hunk 1]
```diff
@@ -23,6 +24,85 @@ pub struct Metadata {
...
+
+    pub group: String,
+    pub version: String,
+    pub scope: String,
+    pub names: CrdNames,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub additionalPrinterColumns: Option<Vec<CrdAdditionalPrinterColumns>>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Default)]
```
Comment (clux) [inline shipcat_definitions/src/crds.rs]: probably shouldn't have default

File: shipcat_definitions/src/crds.rs [hunk 2]
```diff
@@ -23,6 +24,85 @@ pub struct Metadata {
...
+pub struct CrdNames {
+    pub plural: String,
+    pub singular: String,
+    pub kind: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Default)]
```
Comment (clux) [inline shipcat_definitions/src/crds.rs]: also no default

File: shipcat_definitions/src/crds.rs (hunk 1 above)
Comment (JoseAlban) [reply shipcat_definitions/src/crds.rs]: Actually, we need it for `..CrdSpec::default()` ~L70 to infer all other None optionals that might be available.

File: shipcat_definitions/src/crds.rs (hunk 2 above)
Comment (JoseAlban) [reply shipcat_definitions/src/crds.rs]: Same reasoning as above.

**Pull request merged**